[NbConvertApp] Converting notebook 2_train.ipynb to notebook
[NbConvertApp] Executing notebook with kernel: python3
[NbConvertApp] ERROR | Error while converting '2_train.ipynb'
Traceback (most recent call last):
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/nbconvertapp.py", line 410, in export_single_notebook
    output, resources = self.exporter.from_filename(notebook_filename, resources=resources)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/exporters/exporter.py", line 179, in from_filename
    return self.from_file(f, resources=resources, **kw)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/exporters/exporter.py", line 197, in from_file
    return self.from_notebook_node(nbformat.read(file_stream, as_version=4), resources=resources, **kw)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/exporters/notebook.py", line 32, in from_notebook_node
    nb_copy, resources = super(NotebookExporter, self).from_notebook_node(nb, resources, **kw)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/exporters/exporter.py", line 139, in from_notebook_node
    nb_copy, resources = self._preprocess(nb_copy, resources)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/exporters/exporter.py", line 316, in _preprocess
    nbc, resc = preprocessor(nbc, resc)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/preprocessors/base.py", line 47, in __call__
    return self.preprocess(nb, resources)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/preprocessors/execute.py", line 405, in preprocess
    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/preprocessors/base.py", line 69, in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
  File "/home/wesleyy/anaconda3/envs/lv_thrombus/lib/python3.7/site-packages/nbconvert/preprocessors/execute.py", line 448, in preprocess_cell
    raise CellExecutionError.from_cell_and_msg(cell, out)
nbconvert.preprocessors.execute.CellExecutionError: An error occurred while executing the following cell:
------------------
best_model = summary['model obj'][summary['roc_auc'] == max(summary['roc_auc'])][0]
best_technique = summary.index[summary['roc_auc'] == max(summary['roc_auc'])][0]
joblib.dump(best_model,f'pickled_objects/{protocol}_best_model.pkl')
conventional_modified.to_json('results/train_results.csv')
print(f'Best Model: {best_model}')
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-18-dddf8af51314>[0m in [0;36m<module>[0;34m[0m
[1;32m      1[0m [0mbest_model[0m [0;34m=[0m [0msummary[0m[0;34m[[0m[0;34m'model obj'[0m[0;34m][0m[0;34m[[0m[0msummary[0m[0;34m[[0m[0;34m'roc_auc'[0m[0;34m][0m [0;34m==[0m [0mmax[0m[0;34m([0m[0msummary[0m[0;34m[[0m[0;34m'roc_auc'[0m[0;34m][0m[0;34m)[0m[0;34m][0m[0;34m[[0m[0;36m0[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[1;32m      2[0m [0mbest_technique[0m [0;34m=[0m [0msummary[0m[0;34m.[0m[0mindex[0m[0;34m[[0m[0msummary[0m[0;34m[[0m[0;34m'roc_auc'[0m[0;34m][0m [0;34m==[0m [0mmax[0m[0;34m([0m[0msummary[0m[0;34m[[0m[0;34m'roc_auc'[0m[0;34m][0m[0;34m)[0m[0;34m][0m[0;34m[[0m[0;36m0[0m[0;34m][0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 3[0;31m [0mjoblib[0m[0;34m.[0m[0mdump[0m[0;34m([0m[0mbest_model[0m[0;34m,[0m[0;34mf'pickled_objects/{protocol}_best_model.pkl'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m      4[0m [0mconventional_modified[0m[0;34m.[0m[0mto_json[0m[0;34m([0m[0;34m'results/train_results.csv'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m [0mprint[0m[0;34m([0m[0;34mf'Best Model: {best_model}'[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 'protocol' is not defined
NameError: name 'protocol' is not defined

